ikarus/finiteelements/mechanics/kirchhoffloveshell.hh

```diff
@@ -178,40 +164,15 @@ namespace Ikarus {
         energy += (membraneEnergy + bendingEnergy) * geo.integrationElement(gp.position()) * gp.weight();
       }
 
-      if (volumeLoad) {
-        for (const auto& [gpIndex, gp] : uFunction.viewOverIntegrationPoints()) {
-          const auto u                                       = uFunction.evaluate(gpIndex);
-          const Eigen::Vector<double, Traits::worlddim> fExt = volumeLoad(toEigen(geo.global(gp.position())), lambda);
-          energy -= u.dot(fExt) * geo.integrationElement(gp.position()) * gp.weight();
-        }
-      }
+      // External forces volume forces over the domain
+      if (volumeLoad) loads.volume(par, force, dx);
+      energy += static_cast<ScalarType>(force.transpose() * disp);
+      force.setZero();
 
       // line or surface loads, i.e., neumann boundary
       if (not neumannBoundary and not neumannBoundaryLoad) return energy;
-
-      const auto& element = this->localView().element();
-      for (auto&& intersection : intersections(neumannBoundary->gridView(), element)) {
-        if (not neumannBoundary or not neumannBoundary->contains(intersection)) continue;
-
-        const auto& quadLine = Dune::QuadratureRules<double, Traits::mydim - 1>::rule(intersection.type(), order);
-
-        for (const auto& curQuad : quadLine) {
-          // Local position of the quadrature point
-          const Dune::FieldVector<double, Traits::mydim>& quadPos
-              = intersection.geometryInInside().global(curQuad.position());
-
-          const double intElement = intersection.geometry().integrationElement(curQuad.position());
-
-          // The value of the local function
-          const auto u = uFunction.evaluate(quadPos);
-
-          // Value of the Neumann data at the current position
-          const auto neumannValue
-              = neumannBoundaryLoad(toEigen(intersection.geometry().global(curQuad.position())), lambda);
-          energy -= neumannValue.dot(u) * curQuad.weight() * intElement;
-        }
-      }
-
+      loads.traction(par, neumannBoundary, force, dx);
```

Comment:
Can you please remove all load related contributions to this commit, since we discussed to keep PRs smaller. 
This would be not so big but I think we have to refactor the load related part.

Thus, I dont comment on it now and you provide a seperate PR? Sorry :/

Reply:
I have removed `getDisplacementFunction` to a separate PR #223. This PR will thus focus only on load functionalities. After modifications according to the above, I will re-request review and let you know.

Reply:
I have moved everything related to `getDisplacementFunction` to #223. Hope that makes the PR clearer and differentiable.